### PR TITLE
[vcpkg-ci] Request conversion of CONTROL files

### PR DESCRIPTION
**Pull request by autoantwort**
They are often not updated. After this PR this will be requested automatically.

### CODE REVIEW

File: .github/workflows/untrustedPR.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         with:
-          # fetch-depth 2 ensures we capture both parents of the merge commit
-          fetch-depth: 2
+          # fetch-depth 50 tries to ensures we capture the whole history of the branch
```
Comment (Carbenium) [inline .github/workflows/untrustedPR.yml:17]:
```suggestion
          # fetch-depth 50 tries to ensure we capture the whole history of the branch

```

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

**Comment by ras0219-msft** [comment]
Overall this looks great to me; I need to sit down and review the specific git command pipeline before merging but I don't expect any further work needed.

**Comment by BillyONeal** [comment]
Should we consider just bulk doing this instead of asking contributors to do it?

**Comment by autoantwort** [comment]
> Should we consider just bulk doing this instead of asking contributors to do it?

I would prefer this, but it seemed like there was no consensus on this.

**Comment by Neumann-A** [comment]
> I would prefer this, but it seemed like there was no consensus on this.

Maybe it could be done with the next internal CMake update. This will invalidate all binary caches any way.

**Comment by BillyONeal** [comment]
>I would prefer this, but it seemed like there was no consensus on this.

https://github.com/microsoft/vcpkg/pull/19986

**Comment by BillyONeal** [comment]
Do we still want this now that #19986 is merged?

**Comment by autoantwort** [comment]
Hm yeah maybe when someone wants to add an CONTROL file one would get this message.

**Review by BillyONeal — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Can we not remove a bunch of the complexity, given that _any_ CONTROL files are now a problem?

**Comment by autoantwort** [comment]
Yeah we could change `./vcpkg format-manifest ports/*/vcpkg.json` to `./vcpkg format-manifest --all --convert-control`

**Comment by BillyONeal** [comment]
> Yeah we could change `./vcpkg format-manifest ports/*/vcpkg.json` to `./vcpkg format-manifest --all --convert-control`

You can also avoid all the git stuff (in particular, trying to fetch more history) since any remaining CONTROL is doom

**Comment by autoantwort** [comment]
> You can also avoid all the git stuff (in particular, trying to fetch more history) since any remaining CONTROL is doom

Yeah. But I wanted to create another PR where I search for `vcpkg_configure_cmake` and suggest a replacement by `vcpkg_cmake_configure`. Or do you want to do another bulk conversion?

**Comment by BillyONeal** [comment]
> Yeah. But I wanted to create another PR where I search for `vcpkg_configure_cmake` and suggest a replacement by `vcpkg_cmake_configure`. Or do you want to do another bulk conversion?

I'm not sure how I feel about this one. It is a thing that would be nice for contributors to do, but I don't think someone showing up just to bump a version number should be obligated to do it. (It made sense to require folks to update from CONTROL because it was "run this command and that's it" level of investment with us)

Upstream contributors don't want to have to learn the ins and outs of whatever our current best practices are, they just want to make their current version available and get on with their life. Putting it in this check would mark the PR as "request changes" so I'm not sure it's appropriate, although it could be if it's reduced to just a comment "consider adopting vcpkg_cmake_Xxx" or similar.

**Comment by BillyONeal** [comment]
(For what it's worth I don't know if my opinion on not requiring contributors to make such changes is shared by the other maintainers)

**Comment by autoantwort** [comment]
We can choose a neutral message like `if you feel able to to it` and don't mark it as `request changes` when this is the only "Issue"

**Comment by BillyONeal** [comment]
> We can choose a neutral message like `if you feel able to to it` and don't mark it as `request changes` when this is the only "Issue"

Sure.

OK so to land this I think we need at least https://github.com/microsoft/vcpkg/pull/19698#issuecomment-916759299

I would feel better if the detection was also changed to just look for any control files rather than edited ones, but I wouldn't block merging this PR over it.

**Comment by autoantwort** [comment]
@BillyONeal I have created #20142 for the deprecated cmake functions

**Comment by BillyONeal** [comment]
Thanks for the additional helpful text for contributors!

**Pull request merged**